packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -50,30 +47,21 @@ const Category = props => {
         });
     }, [currentPage]);
 
-    if (error) return <div>Data Fetch Error</div>;
-    // If our pagination component has mounted, then we have
-    // a total page count in the store, so we continue to render
-    // with our last known total
-    if (loading || !data)
-        return pageControl.totalPages ? (
-            <CategoryContent pageControl={pageControl} pageSize={pageSize} />
-        ) : (
-            loadingIndicator
-        );
+    useEffect(() => {
+        if (data) {
+            setTotalPages(data.category.products.page_info.total_pages);
+        }
+    }, [data]);
```

Comment:
I think we only need to check this if the `total_pages` change rather than all of the `data` prop. Would prevent an unnecessary re-render if other props change but the total pages stays the same.

Reply:
The initialState of `data = null`, and React's dependency checking didn't seem intelligent enough to do `!== undefined` down the chain, so this wasn't possible w/o triggering a `Cannot read property 'category' of null` error.

Reply:
You could pull out the prop before the hook use:

```js
let total_pages = null;
if (data) {
  { category: { products : { page_info: total_pages } } } } = data;
}
useEffect(() => { setTotalPages(total_pages) }, [total_pages])
```

Reply:
The line `const { total_pages } = data && data.category.products.page_info` might be more readable, but otherwise, that's exactly it. JavaScript deserves a proper coalesce operator and it doesn't have one.

Reply:
This is an alternative.

```js
    const total_pages = data ? data.category.products.page_info.total_pages : null;
    useEffect(() => {
        setTotalPages(total_pages);
    }, [total_pages]);

```

Reply:
I'm not a fan of this change and think y'all are over-optimizing (useState sets with identical values are noop), but made it anyway; I hope you're happy 😭